### DISCUSSION TITLE
Improve UI handling when moving node to rank where node's children already exist

### DISCUSSION THREAD

**Discussion by bronwyncombs** [specify/specify7#5967] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
**From Nate @ Ohio [on the forum](https://discourse.specifysoftware.org/t/specify-7-crashes-when-attempting-to-add-tree-record-from-co-ce-query-dialogs/1856):**
> Encountered this error when trying to fix some records in our geography tree. When working with the tree, I was trying to move a node in the “State” rank to be a child of a node in the same rank.
> 
> I wanted to make “South Wales” (which is not a real geographic place) a child of “Wales”, or alternatively synonymize it with “Pembrokeshire” (since the only record we have in our database from “South Wales” is from the county of Pembrokeshire.)
> 
> The tree wouldn’t allow me to do this (“Can’t move this node here”) I assume because the node has a defined rank and cannot be moved into a rank that differs from the rank it is defined as currently.
> ![image](https://github.com/specify/specify7/assets/135047322/ab657e50-ac6c-48df-9e35-f000f8671f78)
>
> When “use auto generated form” is unchecked, I am unable to change the parent of geo of any record in the geography tree, and I am also unable to enter a value other than the existing value to the Rank picklist.
> I fixed my particular issue just by deleting the node and updating the locality record itself.

---

The problem stems from trying to move a node to a rank where the node's children already exist. However, there is no indication to the user that this is the problem. The only message is the "Bad Tree Structure" warning that appears when hovering over the parent field. 

<img width="1312" alt="Screenshot 2024-07-10 at 11 27 44 AM" src="https://github.com/specify/specify7/assets/135047322/9d30879e-81a8-45da-83f2-5862a4ef075b">

**Describe the solution you'd like**

It would be helpful if more information on what prevents a node from being moved. As noted in the quote above, this user attempted with the move node button before editing the record to achieve the desired placement. 

Instead of "Can't move tree node here", this should be expanded on in some way to inform the user of the limitation/issue

<img width="1272" alt="Screenshot 2024-07-10 at 11 38 34 AM" src="https://github.com/specify/specify7/assets/135047322/9478566f-b3a6-445d-b756-01afa33bf134">

**Reported By**
Nate @ Ohio (Mollusk)

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/specify-7-crashes-when-attempting-to-add-tree-record-from-co-ce-query-dialogs/1856/5
